cylc/flow/subprocctx.py

```diff
@@ -158,12 +158,3 @@ def get_signature(self):
         args = self.func_args + [
             "%s=%s" % (i, self.func_kwargs[i]) for i in skeys]
         return "%s(%s)" % (self.func_name, ", ".join([str(a) for a in args]))
-
-    def __str__(self):
-        return (
-            f'{self.func_name}('
-            f'{", ".join(map(str, self.func_args + list(self.func_kwargs)))}'
-            f'):{self.intvl}'
-        )
-
-    __repr__ = __str__
```

Comment:
I'm struggling to understand how removing this fixes the problem

Reply:
The `__str__` method in the class this one inherits from prints out stdout/err from the xtrigger script (stored in `self.err` and `self.out`). By overriding this we lost that information to the logging.

Reply:
Looks like it's been this way for 4 years according to git blame

Reply:
Don't know if @oliver-sanders would like to keep the `__repr__` method however?

Reply:
Here's the commit which added the `__str__` method: https://github.com/cylc/cylc-flow/commit/c971b9605977f53c989cbfdedb06ce4f1d19ae72

This overrode `SubProcContext.__str__` which was previously being called and includes logic for including stdout/err.

I've got a feeling that the old `__str__` method causes traceback in some situations or something like that. I clearly didn't spot that it was being used in some `LOG.debug` statement and assumed it was an internal interface.

The `__repr__` method was for use in doctests.